### ISSUE TITLE
Support for "gcloud beta emulators datastore"

### ISSUE THREAD

**Issue by broady** [googleapis/google-cloud-common#24]
```
$ gcloud beta emulators datastore env-init
export DATASTORE_HOST=http://localhost:8732
export DATASTORE_DATASET=cbro-scratch
```

See this CL for the implementation to support these two environment variables in Go:
https://code-review.googlesource.com/#/c/3350

I believe this might already have (partial) support in Python?
- [x] Python
- [x] Node (https://github.com/GoogleCloudPlatform/gcloud-node/issues/1040)
- [ ] Golang
- [x] Ruby (GoogleCloudPlatform/gcloud-ruby#237)
- [x] Java

cc @jonparrott

**Comment by dhermes** [comment]
This is 100% [supported][1] in Python. The transport uses the `DATASTORE_HOST` if it is set and the default dataset ID is picked up as `DATASTORE_DATASET` **unless** the user has also set `GCLOUD_DATASET_ID` on their system (production env. var. that we enable). 

If desired, we could switch the precedence for inferring the dataset ID. It is currently:

1. `GCLOUD_DATASET_ID` env. var.
1. `DATASTORE_DATASET` env. var.
1. Determined locally if running in GAE
1. Determined locally if running on GCE

[1]: https://github.com/GoogleCloudPlatform/gcloud-python/blob/6df44a2c0f21bf5fb057f56d28e584dfe87b5c7b/gcloud/environment_vars.py#L30-L34

**Comment by broady** [comment]
How are 3 and 4 determined? We'd like to do the same for Go.

**Comment by jgeewax** [comment]
We had a similar discussion on https://github.com/GoogleCloudPlatform/gcloud-meta/issues/4 which should probably be moved here...

**Comment by dhermes** [comment]
GAE is [done via services][1] provided by the Python runtime in GAE. The Golang equivalent is [`appengine.AppID`][2], but I'm not sure what idiomatic Go equivalent would be to checking if an import succeeded or failed and returning `nil` if the import failed.

GCE is done by [hitting a specific URI][3] common to both GCE and EC2, but using a header which specifies that we want to get metadata for GCE.

[1]: https://github.com/GoogleCloudPlatform/gcloud-python/blob/6df44a2c0f21bf5fb057f56d28e584dfe87b5c7b/gcloud/_helpers.py#L140-L150
[2]: https://cloud.google.com/appengine/docs/go/reference#AppID
[3]: https://github.com/GoogleCloudPlatform/gcloud-python/blob/6df44a2c0f21bf5fb057f56d28e584dfe87b5c7b/gcloud/_helpers.py#L170-L173

**Comment by blowmage** [comment]
I get an error when running `$ gcloud beta emulators datastore env-init`, but I was able to get datastore running using `$ gcloud beta emulators datastore start`.

Once running, I was able to connect to the running datastore instance from a gcloud-ruby instance using the `DATASTORE_HOST` and `DATASTORE_PROJECT` environment variables. We don't support the `DATASTORE_DATASET` environment variable yet but [it is being added](https://github.com/GoogleCloudPlatform/gcloud-ruby/issues/237).
